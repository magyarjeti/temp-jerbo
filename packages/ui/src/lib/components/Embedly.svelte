<script lang="ts">
	import { loadScript } from '@packages/toolbox/loadScript';
	import { onMount, type Snippet } from 'svelte';
	import { getThemeContext } from '../contexts/theme.svelte';
	import { urlRegExp } from '@packages/toolbox/string/url';

	const EMBEDLY_SCRIPT_SRC = 'https://cdn.embedly.com/widgets/platform.js';
	const themeContext = getThemeContext();

	interface Props {
		src?: string | null;
		children?: Snippet;
	}

	let { src, children }: Props = $props();
	let localTheme = $state(themeContext.getPreferredTheme());

	$effect(() => {
		localTheme = themeContext.getPreferredTheme();
	});

	onMount(() => {
		loadScript(EMBEDLY_SCRIPT_SRC);
	});
</script>

{#key localTheme}
	<div>
		<a
			href={src}
			class="embedly-card"
			data-card-via=""
			data-card-chrome="1"
			data-card-analytics="1"
			data-card-theme={localTheme}
			data-card-controls="0"
			data-card-width=""
			data-card-recommend="0"
		>
			{@render children?.()}
		</a>
	</div>
{/key}
