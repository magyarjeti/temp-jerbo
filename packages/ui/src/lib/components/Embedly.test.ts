import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import Embedly from './Embedly.svelte';

test('Embedly snapshot', () => {
	const { container } = render(Embedly, { props: { src: 'https://444.hu' } });
	expect(container).toMatchSnapshot();
});
