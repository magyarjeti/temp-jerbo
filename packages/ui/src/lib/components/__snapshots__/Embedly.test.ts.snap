// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Embedly snapshot 1`] = `
<div>
  <a
    class="embedly-card"
    data-card-analytics="1"
    data-card-chrome="1"
    data-card-controls="0"
    data-card-recommend="0"
    data-card-via=""
    data-card-width=""
    href="https://444.hu"
  />
</div>
`;
